public/app/features/dashboard/components/TransformationsEditor/TransformationFilter.tsx

```diff
@@ -2,41 +2,69 @@ import { css } from '@emotion/css';
 import React, { useMemo } from 'react';
 
 import {
-  DataFrame,
   DataTransformerConfig,
   GrafanaTheme2,
+  SelectableValue,
   StandardEditorContext,
   StandardEditorsRegistryItem,
 } from '@grafana/data';
-import { Field, useStyles2 } from '@grafana/ui';
+import { DataTopic } from '@grafana/schema';
+import { Field, Select, useStyles2 } from '@grafana/ui';
 import { FrameSelectionEditor } from 'app/plugins/panel/geomap/editor/FrameSelectionEditor';
 
+import { TransformationData } from './TransformationsEditor';
+
 interface TransformationFilterProps {
   index: number;
   config: DataTransformerConfig;
-  data: DataFrame[];
+  data: TransformationData;
   onChange: (index: number, config: DataTransformerConfig) => void;
 }
 
+const dataFramesSourceOpts: Array<SelectableValue<DataTopic>> = [
+  { value: DataTopic.Series, label: 'Series' },
+  { value: DataTopic.Annotations, label: 'Annotations' },
+];
+
 export const TransformationFilter = ({ index, data, config, onChange }: TransformationFilterProps) => {
   const styles = useStyles2(getStyles);
   const context = useMemo(() => {
     // eslint-disable-next-line
-    return { data } as StandardEditorContext<unknown>;
+    return { data: data.series } as StandardEditorContext<unknown>;
   }, [data]);
 
   return (
-    <div className={styles.wrapper}>
-      <Field label="Apply transformation to">
-        <FrameSelectionEditor
-          value={config.filter!}
-          context={context}
-          // eslint-disable-next-line
-          item={{} as StandardEditorsRegistryItem}
-          onChange={(filter) => onChange(index, { ...config, filter })}
-        />
-      </Field>
-    </div>
+    <>
+      <div>
```

Comment:
need this extra div?

Reply:
what are you, the `<div>` Sheriff? :rofl:

Reply:
🤠